GRDB/Record/PersistableRecord.swift

```diff
@@ -787,6 +789,33 @@ final class DAO<Record: MutablePersistableRecord> {
             databaseTableName: databaseTableName,
             key: key)
     }
+    
+    private struct GroupedPrimaryKeysAndTheirValues {
+        let allColumns: [String]
+        var nullColumns: [String]
+        var notNullColumns: [String]
+        var notNullValues: [DatabaseValue]
+    }
+    
+    // Some primary key columns may be null, in which case they need to be handled differently
+    private func groupedPrimaryKeysAndTheirValues() -> GroupedPrimaryKeysAndTheirValues {
```

Comment:
Consider returning a tuple with two values: column information and the array of database values. This will allow us to send column information to SQL generation methods, without giving database values as well.

Document what this function does. And rename it (after writing the documentation, so that you can read proper English that will inspire you a much better name).